test/unit/utils-nypl-source-mapper.test.js

```diff
@@ -1,77 +1,58 @@
 const expect = require('chai').expect
 const nock = require('nock')
+
+const { stubNyplSourceMapper, unstubNyplSourceMapper } = require('./utils')
+
 const NyplSourceMapper = require('../../lib/utils/nypl-source-mapper')
 
-describe('NyplSourceMapper', async function () {
-  describe('nyplSourceMapping', async function () {
-    const response = {
-      'sierra-nypl': {
-        organization: 'nyplOrg:0001',
-        bibPrefix: 'b',
-        holdingPrefix: 'h',
-        itemPrefix: 'i'
-      },
-      'recap-pul': { organization: 'nyplOrg:0003', bibPrefix: 'pb', itemPrefix: 'pi' },
-      'recap-cul': { organization: 'nyplOrg:0002', bibPrefix: 'cb', itemPrefix: 'ci' },
-      'recap-hl': { organization: 'nyplOrg:0004', bibPrefix: 'hb', itemPrefix: 'hi' }
-    }
-    let callCount = 0
-
-    before(() => {
+const SOURCE_MAPPING_URL = 'https://raw.githubusercontent.com/NYPL/nypl-core/master/mappings/recap-discovery/nypl-source-mapping.json'
+
+describe('utils/NyplSourceMapper', async function () {
+  before(() => NyplSourceMapper.__resetInstance())
+
+  describe('instance', async function () {
+    beforeEach(stubNyplSourceMapper)
+    afterEach(() => {
+      unstubNyplSourceMapper()
       NyplSourceMapper.__resetInstance()
-      nock('https://raw.githubusercontent.com/NYPL/nypl-core/master/mappings/recap-discovery/nypl-source-mapping.json')
-        .defaultReplyHeaders({
-          'access-control-allow-origin': '*',
-          'access-control-allow-credentials': 'true'
-        })
-        .get(/.*/)
-        .reply(200, () => {
-          callCount += 1
-          return response
-        })
     })
-    after(() => NyplSourceMapper.__resetInstance())
 
-    it('should fetch data from nypl core if not initialized', async function () {
+    it('should fetch data from nypl core', async function () {
       const mapping = await NyplSourceMapper.instance()
-      expect(mapping.nyplSourceMap).to.deep.equal(response)
-      expect(callCount).to.equal(1)
+      expect(mapping.nyplSourceMap).to.nested.include({ 'sierra-nypl.organization': 'nyplOrg:0001' })
     })
 
     it('should return pre-fetched data if initialized', async function () {
       const mapping = await NyplSourceMapper.instance()
-      expect(mapping.nyplSourceMap).to.deep.equal(response)
-      expect(callCount).to.equal(1)
+      expect(mapping.nyplSourceMap).to.nested.include({ 'sierra-nypl.organization': 'nyplOrg:0001' })
+
+      // Trigger another instance creation, which will break if another `fetch`
```

Comment:
I'd love to see this as an explicit assertion rather than implicit in the setup of the nock. Can you replace the callback counter from the original testing setup?

Reply:
Sure thing. Added a counter and updated test to make explicit assertions about counts throughout this test.